[Config] Do not use absolute path when computing the vendor freshness

lyrixx opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

When one uses Docker with a different mounting point between CLI & FPM,
the cache keeps regenerating because the ComposerResource class see a
different path for each SAPI. For example `/home/app/app/vendor` vs `/var/www/app/vendor`.
So if you hit FPM, then the CLI, then FPM, each time a new cache is
generated. So the application is quite slow in dev env. And for people
on MacOSX (with docker) is a big pain! And obvisouly, this never
stabilizes !

This occurs a lot when you have a worker, that crash and reboot in the
background, and you browse the web interface. Or when you have something
that hit your API every X secondes, and you are working on a worker.

nicolas-grekas commented on src/Symfony/Component/Config/Resource/ComposerResource.php:
```diff
@@ -71,12 +71,14 @@ private static function refresh()
                 $v = \dirname(\dirname($r->getFileName()));
                 if (file_exists($v.'/composer/installed.json')) {
-                    self::$runtimeVendors[$v] = @filemtime($v.'/composer/installed.json');
+                    self::$runtimeVendors["__vendor_$i"] = @filemtime($v.'/composer/installed.json');
```
This break the getVendors method currently.

lyrixx replied:
Better ?

lyrixx committed:
[Config] Do not use absolute path when computing the vendor freshness

When one uses Docker with a different mounting point between CLI & FPM,
the cache keeps regenerating because the ComposerResource class see a
different path for each SAPI. For example `/home/app/app/vendor` vs
`/var/www/app/vendor`.
So if you hit FPM, then the CLI, then FPM, each time a new cache is
generated. So the application is quite slow in dev env. And for people
on MacOSX (with docker) is a big pain! And obvisouly, this never
stabilizes !

This occurs a lot when you have a worker, that crash and reboot in the
background, and you browse the web interface. Or when you have something
that hit your API every X secondes, and you are working on a worker.

lyrixx commented:
Sorry, I pushed on symfony/symony instead of my fork. 
I reopened it here: https://github.com/symfony/symfony/pull/32579
